tensorflow/python/keras/mixed_precision/experimental/autocast_variable.py

```diff
@@ -188,61 +188,88 @@ def initial_value(self):
   def constraint(self):
     return self._variable.constraint
 
+  def _apply_assign_update(
+      self, update_fn, value, use_locking=None, name=None, read_value=True):
+    if not read_value:
+      return update_fn(value, use_locking, name, read_value)
+
+    if context.executing_eagerly() or ops.inside_function():
```

Comment:
I think `op.executing_eagerly_outside_functions()` should be used instead of these checks, since I'm unsure if this codepath is correct in graph mode inside a function.

And same for `_apply_update`

Reply:
[This test](https://github.com/lgeiger/tensorflow/blob/fix-distributed-autocast-var-assign/tensorflow/python/keras/mixed_precision/experimental/autocast_variable_test.py#L346-L358) runs using tf functions in v1 session mode. If I switch to `ops.executing_eagerly_outside_functions()` it wouldn't work anymore. 
Though, I am happy to not support this usage if there are concerns about the correctness there.

Reply:
Sounds good :+1:
I think `tf.function` running in graph mode is not a very common use case anyway so I agree that it wouldn't justify different functionality here.

`@test_util.run_v2_only` didn't really work with the parameterized test cases, so I had to manually skip the test in f5a0fdaa0aeff548623811b887c6da34303ab25f